hbase-client/src/main/java/org/apache/hadoop/hbase/client/ScannerCallableWithReplicas.java

```diff
@@ -72,11 +73,12 @@ class ScannerCallableWithReplicas implements RetryingCallable<Result[]> {
 
   public ScannerCallableWithReplicas(TableName tableName, ClusterConnection cConnection,
     ScannerCallable baseCallable, ExecutorService pool, int timeBeforeReplicas, Scan scan,
-    int retries, int readRpcTimeout, int scannerTimeout, int caching, Configuration conf,
-    RpcRetryingCaller<Result[]> caller) {
+    int retries, int readRpcTimeout, int scannerTimeout, boolean useScannerTimeoutForNextCalls,
```

Comment:
Is it better to thread through the instance of `ConnectionConfiguration` rather than this new boolean flag? Probably many of these fields can be folded down into an inexpensive field lookup from the `ConnectionConfiguration` instance instead.

Probably this is more of a refactor than you were ready to bite off, and all this code gets tossed as of 3.0, yes?

Reply:
Actually, it's not so simple here. The readRpcTimeout and scannerTimeout both can be overridden on a per-Table basis via TableBuilder. Caching can be overridden by the Scan. So really just retries and my new boolean could be unified in this way. I could do it, but yea trying not to refactor too much. Also I could imagine making retries configurable via TableBuilder in the future which would negate this change... not necessarily somethign we should design for, just saying.

Yes, this code goes away in 3.0, but I'm sure we'll be maintaining it for years to come.